plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptions/Utils.php

```diff
@@ -84,13 +84,20 @@ public static function get_quantity_input_args( $product ) {
 	}
 
 	/**
-	 * Make the quantity input interactive by wrapping it with the necessary data attribute.
+	 * Make the quantity input interactive by wrapping it with the necessary data attribute and adding an input event listener.
 	 *
 	 * @param string $quantity_html The quantity HTML.
 	 * @param string $wrapper_attributes Optional wrapper attributes.
 	 * @return string The quantity HTML with interactive wrapper.
 	 */
 	public static function make_quantity_input_interactive( $quantity_html, $wrapper_attributes = '' ) {
+		$html = new \WP_HTML_Tag_Processor( $quantity_html );
+		while ( $html->next_tag( 'input' ) ) {
+			$html->set_attribute( 'data-wp-on--input', 'actions.handleInputQuantityChange' );
+		}
```

Comment:
So we are adding the attribute to every input element. I can see we only have one input here so it's probably fine. But I wonder if we should inject the directive to only one number input to prevent possible side effects? The `global/quantity-input.php` can be overridden; so in theory, it can have more than one input element.

Reply:
Or I'm just too defensive 😂

Reply:
> Or I'm just too defensive 😂 

Not at all, good call! I updated the check in 6efec239498437b998b035a2e19507f42ca2c51e, so it's only applied to the first `input` with `type="number"` and `name` starting with `quantity`.